deps/rabbit/src/rabbit_msg_store.erl

```diff
@@ -1976,9 +2016,7 @@ load_and_vacuum_message_file(File, State = #gc_state{ dir = Dir }) ->
                       Acc;
                   #msg_location { file = File, total_size = TotalSize,
                                   offset = Offset } = Entry ->
-                      [ Entry | Acc ];
-                  _ ->
-                      Acc
+                      [ Entry | Acc ]
```

Comment:
I have removed this clause as I believe it was hiding the real bug and we should really crash if the data we are getting in the file does not match our in-memory state.

Reply:
The clause is not fully removed anymore, instead we match on what we expect exactly.